docs/glossary.md

```diff
@@ -2,16 +2,20 @@
 
 **Playlist**: This is a representation of an HLS or DASH manifest.
 
+**Media Playlist**: This is a manifest that represents a single rendition of the source.
+
 **Master Playlist Controller**: This acts as the main controller for the playback engine. It interacts with the SegmentLoaders, PlaylistLoaders, PlaybackWatcher, etc.
 
-**Playlist Loader**: This will request the source and load the master manifest. It also interacts with the ABR algorithm to pick a media playlist or wraps a media playlist if it is provided as the source. There are more details about the playlist loader [here](./arch.md).
+**Playlist Loader**: This will request the source and load the master manifest. It is also instructed by the ABR algorithm to load a media playlist or wraps a media playlist if it is provided as the source. There are more details about the playlist loader [here](./arch.md).
 
-**Media Playlist**: This is a manifest that represents a single rendition of the source.
+**Segment Loader**: This determines which segment should be loaded via the Media Segment Loader, decrypts the segment if necessary and passes the result to the Source Updater.
```

Comment:
Might read a bit easier as "This determines which segment should be loaded, requests it via the Media Segment Loader, and passes the result to the Source Updater." (Also, Media Segment Request will decrypt the bytes, but that was my comment's fault.)